src/Elliptic/DiscontinuousGalerkin/Actions/ApplyOperator.hpp

```diff
@@ -0,0 +1,629 @@
+                  temporal_id, std::move(mortar_data));
+            }
+          });
+    }
+
+    // Apply DG operator
+    db::mutate<OperatorAppliedToFieldsTag, all_mortar_data_tag>(
+        make_not_null(&box),
+        [](const auto operator_applied_to_fields, const auto all_mortar_data,
+           const auto& primal_fields, const auto& primal_fluxes,
+           const auto&... args) noexcept {
+          elliptic::dg::apply_operator<System, Linearized>(
+              operator_applied_to_fields, all_mortar_data, primal_fields,
+              primal_fluxes, args...);
+        },
+        db::get<PrimalFieldsTag>(box), db::get<PrimalFluxesTag>(box),
+        db::get<domain::Tags::Mesh<Dim>>(box),
+        db::get<domain::Tags::InverseJacobian<Dim, Frame::Logical,
+                                              Frame::Inertial>>(box),
+        db::get<domain::Tags::Interface<
+            domain::Tags::InternalDirections<Dim>,
+            ::Tags::Magnitude<domain::Tags::UnnormalizedFaceNormal<Dim>>>>(box),
+        db::get<domain::Tags::Interface<
+            domain::Tags::BoundaryDirectionsInterior<Dim>,
+            ::Tags::Magnitude<domain::Tags::UnnormalizedFaceNormal<Dim>>>>(box),
+        db::get<::Tags::Mortars<domain::Tags::Mesh<Dim - 1>, Dim>>(box),
+        db::get<::Tags::Mortars<::Tags::MortarSize<Dim - 1>, Dim>>(box),
+        db::get<elliptic::dg::Tags::PenaltyParameter>(box), temporal_id,
+        std::forward_as_tuple(db::get<SourcesArgsTags>(box)...));
+
+    return {std::move(box)};
+  }
+};
+
+}  // namespace detail
+
+/*!
+ * \brief Initialize DataBox tags for applying the DG operator
+ *
+ * \see elliptic::dg::Actions::apply_operator
+ */
+template <typename System>
+using initialize_operator =
+    tmpl::list<detail::InitializeFacesAndMortars<System>>;
+
+/*!
+ * \brief Apply the DG operator to the `PrimalFieldsTag` and write the result to
+ * the `OperatorAppliedToFieldsTag`
+ *
+ * Add this list to the action list of a parallel component to compute the
+ * elliptic DG operator or its linearization. The operator involves a
+ * communication between nearest-neighbor elements. See `elliptic::dg` for
+ * details on the elliptic DG operator. Make sure to add
+ * `elliptic::dg::Actions::initialize_operator` to the initialization phase of
+ * your parallel component so the required DataBox tags are set up before
+ * applying the operator.
+ *
+ * The result of the computation is written to the `OperatorAppliedToFieldsTag`.
+ * Additionally, the primal fluxes are written to the `PrimalFluxesTag` as an
+ * intermediate result. The auxiliary fields and fluxes are discarded to avoid
+ * inflating the memory usage.
+ *
+ * You can specify the `PrimalMortarFieldsTag` and the `PrimalMortarFluxesTag`
+ * to re-use mortar-data memory buffers from other operator applications, for
+ * example when applying the nonlinear and linearized operator. They default to
+ * the `PrimalFieldsTag` and the `PrimalFluxesTag`.
```

Comment:
What is the downside to using the default?  Does it overwrite the primal fluxes mentioned in the previous paragraph?

Reply:
The linear and nonlinear operator apply to different variables, so when using the defaults here they each set up memory buffers for their mortar data. However, when the operator applications don't overlap in the algorithm, as is the case for the Newton-Krylov algorithm, we can just re-use the same buffers in the DataBox for both the linear and nonlinear operator. In that case we can set these template parameters to the same tags.